src/main/java/com/chessix/sepa/pain/impl/Pain00100103.java

```diff
@@ -74,7 +78,7 @@ private void initializeDocument() {
 
 		// Payment Information
 		PaymentInstructionInformation3 pi = new PaymentInstructionInformation3();
-		pi.setPmtInfId(DocumentUtils.createUniqueId());
+		pi.setPmtInfId(paymentInfoId);
```

Comment:
Indeed, the paymentInfoId is created by the DocumentServiceImpl. So if none was given, it is autogenerated like it always was.